coral/src/app/features/requests/components/schemas/components/SchemaRequestTable.tsx

```diff
@@ -0,0 +1,132 @@
+import { DataTable, DataTableColumn, EmptyState } from "@aivenio/aquarium";
+import infoIcon from "@aivenio/aquarium/dist/src/icons/infoSign";
+import deleteIcon from "@aivenio/aquarium/dist/src/icons/delete";
+import {
+  requestStatusChipStatusMap,
+  requestStatusNameMap,
+} from "src/app/features/approvals/utils/request-status-helper";
+import {
+  requestOperationTypeChipStatusMap,
+  requestOperationTypeNameMap,
+} from "src/app/features/approvals/utils/request-operation-type-helper";
+import { SchemaRequest } from "src/domain/schema-request";
+
+interface SchemaRequestTableRow {
+  deletable: SchemaRequest["deletable"];
+  environment: SchemaRequest["environmentName"];
+  id: SchemaRequest["req_no"];
+  requestStatus: SchemaRequest["requestStatus"];
+  requestType: SchemaRequest["requestOperationType"];
+  requestedBy: SchemaRequest["username"];
+  requestedOn: SchemaRequest["requesttimestring"];
+  schemaversion: SchemaRequest["schemaversion"];
+  topic: SchemaRequest["topicname"];
+}
+
+type SchemaRequestTableProps = {
+  requests: SchemaRequest[];
+};
+
+function SchemaRequestTable({ requests }: SchemaRequestTableProps) {
+  const columns: Array<DataTableColumn<SchemaRequestTableRow>> = [
+    { type: "text", field: "topic", headerName: "Topic" },
+    {
+      type: "status",
+      field: "environment",
+      headerName: "Environment",
+      status: ({ environment }) => ({
+        status: "neutral",
+        text: environment,
+      }),
+    },
+    {
+      type: "status",
+      field: "requestStatus",
+      headerName: "Status",
+      status: ({ requestStatus }) => {
+        return {
+          status: requestStatusChipStatusMap[requestStatus],
+          text: requestStatusNameMap[requestStatus],
+        };
+      },
+    },
+    {
+      type: "status",
+      field: "requestType",
+      headerName: "Type",
+      status: ({ requestType }) => {
+        return {
+          status: requestOperationTypeChipStatusMap[requestType],
+          text: requestOperationTypeNameMap[requestType],
+        };
+      },
+    },
+    { type: "text", field: "requestedBy", headerName: "Requested by" },
+    {
+      type: "text",
+      field: "requestedOn",
+      headerName: "Requested on",
+      formatter: (value) => {
+        return `${value}${"\u00A0"}UTC`;
+      },
+    },
+    {
+      type: "action",
+      headerName: "Details",
+      headerInvisible: true,
+      width: 30,
+      action: ({ id }) => ({
+        text: "View",
+        icon: infoIcon,
+        onClick: () => console.log("details!", id),
+      }),
```

Comment:
Hey, it appears (while not in the type definitions) you can return button props quite freely as the return value of the `action` function. For example `loading` works while not suggested.

It should work for aria attributes as well:

```
action: ({ id }) => ({
        text: "View",
        icon: infoIcon,
        onClick: () => onDetails(String(id)),
        'aria-label': `View request ${id} details`
      }),

```
![image](https://user-images.githubusercontent.com/15416578/225438395-f6e08af4-9aaf-48a0-a8bc-ac4cda8c99c8.png)


I believe this would help providing more descriptive label for the screen reader users and making them more targetable in tests as well.

Reply:
Oh, that's awesome 🎉  Great find! Will update my PR.